Neos.Fusion.Form/Classes/Eel/FormHelper.php

```diff
@@ -0,0 +1,115 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Fusion\Form\Eel;
+
+/*
+ * This file is part of the Neos.Fusion.Form package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Eel\ProtectedContextAwareInterface;
+use Neos\Flow\Security\Context as SecurityContext;
+use Neos\Flow\Security\Cryptography\HashService;
+use Neos\Flow\Mvc\Controller\MvcPropertyMappingConfigurationService;
+
+class FormHelper implements ProtectedContextAwareInterface
+{
+    /**
+     * @Flow\Inject
+     * @var SecurityContext
+     */
+    protected $securityContext;
+
+    /**
+     * @Flow\Inject
+     * @var MvcPropertyMappingConfigurationService
+     */
+    protected $mvcPropertyMappingConfigurationService;
+
+    /**
+     * @Flow\Inject
+     * @var HashService
+     */
+    protected $hashService;
+
+    /**
+     * Calculate the trusted properties token for the given form content
+     *
+     * @param array $arguments
+     * @param string|null $fieldNamePrefix
+     */
+    public function argumentsWithHmac(array $arguments = [], string $excludeNamespace = '')
+    {
+        if ($excludeNamespace !== null && isset($arguments[$excludeNamespace])) {
+            unset($arguments[$excludeNamespace]);
+        }
+        return $this->hashService->appendHmac(base64_encode(serialize($arguments)));
+    }
+
+    /**
+     * Calculate the trusted properties token for the given form content
+     *
+     * @param string $content
+     * @param string|null $fieldNamePrefix
+     */
+    public function trustedPropertiesToken(string $content, string $fieldNamePrefix = '')
+    {
+        $dom = new \DOMDocument();
+        $dom->loadHTML($content);
```

Comment:
Hey, 

I think parsing this is not really stable / feasible. I just wrapped the $dom->loadHTML() with a try/catch block (because it did not work in my case); and I got the error 
`Warning: DOMDocument::loadHTML(): htmlParseEntityRef: no name in Entity, line: 1 in .../Data/Temporary/Development/Cache/Code/Flow_Object_Classes/Neos_Fusion_Form_Eel_FormHelper.php line 66`

So I'd rather suggest we add a registerFieldName() method to the FormDefinition object, which can be called from the field.

What do you think? :)
All the best, Sebastian

Reply:
I pushed a better version. The dom parser can behave much more relaxed like in the htmlAugmenter.

I actually like the parsing approach as it handles all fields equal regardless wether they were created by the form prototypes or are pure afx.